File: backend/src/main/java/com/tuankhoi/backend/dto/response/UserResponse.java
package com.tuankhoi.backend.dto.response;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.io.Serializable;
import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class UserResponse implements Serializable {
    String id;

    String userName;

    String password;

    String email;

    String fullName;

    String imageUrl;

    String cloudinaryImageId;

    @Builder.Default
    boolean active = true;

    String createdBy;

    @JsonSerialize(using = LocalDateTimeSerializer.class)
    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
    LocalDateTime createdDate;

    String lastModifiedBy;

    @JsonSerialize(using = LocalDateTimeSerializer.class)
    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
    LocalDateTime lastModifiedDate;

    RoleResponse role;
}

File: frontend/src/app/api/model/response/user-response.ts
import { RoleResponse } from "./role-response";

export interface UserResponse {
    id: string;
    userName: string;
    password: string;
    email: string;
    fullName: string;
    imageUrl: string;
    cloudinaryImageId: string;
    active: boolean;
    createdBy: string;
    createdDate: Date | string;
    lastModifiedBy: string;
    lastModifiedDate: Date | string;
    role: RoleResponse;
}
